### DISCUSSION TITLE
3.6.24之后版本，组件不在src目录下ts type报错

### DISCUSSION THREAD

**Discussion by xiaof520** [NervJS/taro#15753] [Q & A] [answered]
### 这个特性解决了什么问题？
有没有配置可以自定义某些非src目录支持ts

**Comment by xiaof520** [comment]
好像和pnpm workspace   symlinks有关， ln -s 的文件夹有问题，，

**Comment by xiaof520** [comment]
resolve symlinks: true  不知道是不是解析真实地址影响，pnpm workspace   ln -s  npm link 真实地址不在src下 ts type报错

**Comment by eiinu** [comment]
同样的问题，通过 alias 配置引用了项目根目录以外的文件，在 dev 和 build 时都会报 ts 类型错误
3.6.24 正常，之后的版本出现了问题

**Comment by ZakaryCode** [comment, marked as answer]
把文件加入到 tsconfig 的 include 内

**Comment by eiinu** [comment]
准确地说，不是 TS 类型错误，是 TS 文件没被 babel-loader 编译，当成 JS 模块加载，有语法识别不了的报错。
所以需要把目标文件加入 Taro config 的 `h5.compile.include`、`mini.compile.include` 里面。
原因参考这个：https://github.com/NervJS/taro/issues/15339

**Reply by xiaof520** [reply]
谢谢， 按你的可以了， 之前尝试过改tsconfig ，module.rules.include  use babel-loader都不行，，新版可以用了
